dwave_networkx/drawing/draw_parallel_embeddings.py

```diff
@@ -0,0 +1,242 @@
+# Copyright 2024 D-Wave
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+from typing import Union, List, Dict, Tuple, Optional
+import random
+import math
+
+import networkx as nx
+
+from dwave_networkx.drawing.chimera_layout import draw_chimera, chimera_layout
+from dwave_networkx.drawing.pegasus_layout import draw_pegasus, pegasus_layout
+from dwave_networkx.drawing.zephyr_layout import draw_zephyr, zephyr_layout
+
+def _generate_node_color_dict(
+    G: nx.Graph,
+    embeddings: List[dict],
+    S: nx.Graph = None,
+    one_to_iterable: bool = False,
+    shuffle_colormap: bool = True,
+    seed: Optional[int] = None,
+) -> Tuple[Dict, List[dict]]:
+    """Generate a node color dictionary mapping each node in G to an embedding index or NaN.
+
+    Args:
+        G: The target graph.
+        embeddings: A list of embeddings (each embedding a dict from source nodes to target nodes).
+        S: The optional source graph.
+        one_to_iterable: If True, a single source node maps to multiple target nodes.
+        shuffle_colormap: If True, embeddings are shuffled before assigning colors.
+        seed: A seed for the random number generator.
+
+    Returns:
+        node_color_dict: A dictionary mapping each node in G to either an embedding index or NaN.
+        _embeddings: The potentially shuffled embeddings list used for assigning colors.
+    """
+    node_color_dict = {q: float("nan") for q in G.nodes()}
+
+    if shuffle_colormap:
+        _embeddings = embeddings.copy()
```

Comment:
Minor, but I'd name the embeddings copy to `embeddings_` rather than with an `_` at the start (which signifies it being private, which isn't really applicable here).